### PR TITLE
Add module interface documentation

### PR DESCRIPTION
**Pull request by NitenIchiry**
Add module interface documentation and add link to documentation in GtModuleInterface.

## Checklist:

- [ ] A test for the new functionality was added (if applicable).
- [ ] All tests run without failure.
- [ ] The changelog has been extended, if this MR contains important changes from the users's point of view.
- [ ] The new code complies with the GTlab's style guide.
- [ ] New interface methods / functions are exported via `EXPORT`. Non-interface functions are NOT exported.
- [ ] The number of code quality warnings is not increasing.

### CODE REVIEW

**Comment by rainman110** [comment]
@NitenIchiry Great! Thank you for you work. Could you please upload the missing image? I think you just forgot to add  it to git.

I will keep the issue open such that we can continue working on the documentation in this branch.

**Comment by NitenIchiry** [comment]
Yes, that would be ideal. DO you suggestions for updating the .gitignore ? The path docs/modules was ignored because of the field /modules in the ignore file.

**Comment by rainman110** [comment]
> Yes, that would be ideal. DO you suggestions for updating the .gitignore ? The path docs/modules was ignored because of the field /modules in the ignore file.

Just add `!docs/modules/`

**Comment by NitenIchiry** [comment]
Done. I guess we can merge this now.

**Comment by NitenIchiry** [comment]
> As is have written above, let's keep the MR open to add more documentation before merging 😉 
> 
> Thanks for your work !

Yeah good point, keeping the issue open doesn't make a lot of sense 🙂.

**Comment by NitenIchiry** [comment]
@rainman110 I added documentation for exporter and importer interfaces. Maybe you can look over these and correct any mistakes I made. I also made afirst version for the mdi documentation. But I was not really sure about all of the virtual function. So you can improve on my first outline. I did not get to write about all of the interfaces. Mostly because there also is not any documentation yet in the wiki.

**Comment by rainman110** [comment]
@jensschmeink I think we should keep tasks and calculators on one page.

File: docs/basics/data_modelling.rst [hunk 1]
```diff
@@ -1,15 +1,111 @@
...
+To perform a recursive query, including sub-objects, replace ``findDirectChild`` with ``findChild`` etc.
+
+Query child objects  by UUID or path
+""""""""""""""""""""""""""""""""""""
+
+To retrieve objects using unique identifiers (UUIDs) or hierarchical paths using both direct and recursive searches,
+the functions :cpp:func:`GtObject::getObjectByUuid()` and :cpp:func:`GtObject::getObjectByPath()` are used.
+
+
+Adding child objects
```
Comment (jensschmeink) [inline docs/basics/data_modelling.rst:59]: @rainman110 We have to keep in mind (and maybe should mention), that there are cases in which the gtDatamodel has to be used to add a child correctly.

Comment (rainman110) [reply docs/basics/data_modelling.rst:59]: Good point. I have no clue, when it should be used though and when not. Maybe @real-ct-ac can clarify?

**Comment by jensschmeink** [comment]
> @jensschmeink I think we should keep tasks and calculators on one page.

done

File: docs/basics/data_modelling.rst (hunk 1 above)
Comment (jensschmeink) [reply docs/basics/data_modelling.rst:59]: It is always needed if the add is called in the GUI  (as in UI elements) and not via calculator, importer

Comment (rainman110) [reply docs/basics/data_modelling.rst:59]: What is the difference? I am sure I use appendChild also from GUI commands (e.g. when importing something) and it also seems to work.

Comment (jensschmeink) [reply docs/basics/data_modelling.rst:59]: I think the main difference is the update of the model in the GUI.

Comment (rainman110) [reply docs/basics/data_modelling.rst:59]: Is it safe to just always use the datamodel->appendChild() function? If so, we should only document this.

Comment (jensschmeink) [reply docs/basics/data_modelling.rst:59]: No. In calculators it should not be used.

**Pull request merged**